service/src/main/java/io/nuvalence/ds4g/documentmanagement/service/processor/impl/DocumentAiProcessor.java

```
@@ -93,15 +96,17 @@ public DocumentProcessorResult process(String documentId) {
 
             R customParsedResult = parseResults(result);
 
-            return response.documentId(UUID.fromString(documentId))
+            return responseBuilder
                     .status(DocumentProcessorStatus.COMPLETE)
                     .result(customParsedResult)
                     .timestamp(OffsetDateTime.now())
                     .build();
-        } catch (IOException e) {
+        } catch (Exception e) {
```

you may be right in most cases, that if an Exception happens , processors cannot be run on document.
Just consider when for some reason processors are not available, due to an issue at GCP/configuration or a network issue problem .
In case of network problems , there are timeout exception , in those cases, probably in a few other, the best is to retry and returning "UNPROCESSABLE" will not allow it

---

I really thought about this as well. And processors have been made to be re triggable... If a processor fails it can be re triggered. 
I really believe this is a good fix for a bug. And we could see if we really want to have retry. Or allow a processor to be re triggered on demand. If there are retries anyway we would need to retrigger it manually if that continues failing. So implementing a retry approach based on certain errors may be a new feature rather than a bug fix.
Since the result of the processor is a helping tool for the agent. We could allow the agent to retrigger the processor if needed.
What do you think?

---

@JayStGelais any thoughts on this? I'm not sure how I feel yet about the additional status. I can see pros and cons.

---

I think this change is fine as written as it captures additional insight into why the processor failed.

That said, I think we should capture a story for retries, using the topic retry policy to control timing and number of tries. We would want to examine the specific exceptions thrown and evaluate whether they are expected to be retryable and fail fast on known unprocessable exceptions (such as invalid arguments like this case).